### PR TITLE
Enable unprivileged_userfaultfd on centos 9 to support post-copy migration

### PR DESCRIPTION
**Pull request by iholder101**
For more info, please look at
https://github.com/openshift/machine-config-operator/pull/3724

### CODE REVIEW

File: cluster-provision/centos9/Dockerfile [hunk 1]
```diff
@@ -32,3 +32,7 @@ RUN curl -L -o /initrd.img http://mirror.stream.centos.org/9-stream/BaseOS/x86_6
 RUN curl -L -o /vmlinuz http://mirror.stream.centos.org/9-stream/BaseOS/x86_64/os/images/pxeboot/vmlinuz
 
 COPY scripts/* /
+
+# Enable userfaultfd for non-root users to enable post-copy migration.
+# For more info, please look at: https://github.com/openshift/machine-config-operator/pull/3724
+RUN sysctl vm.unprivileged_userfaultfd=1
```
Comment (rmohr) [inline cluster-provision/centos9/Dockerfile:38]: I think you want to run this command in the actual guest, so you have to move it to the provision scripts (check for `sysctl`, we should have some commands already at the corresponding places.

Comment (iholder101) [reply cluster-provision/centos9/Dockerfile:38]: Thank you @rmohr.
The thing is that we probably want to run this only on centos 9, and I'm not sure what's the right place to put this

Comment (rmohr) [reply cluster-provision/centos9/Dockerfile:38]: you can probably check in `kubevirtci/cluster-provision/k8s/1.27/k8s_provision.sh` for the OS the script is executed on. Not sure what's the simples way but `lsb_release`, `/etc/centos-version`, ... would be options to explore.

Comment (rmohr) [reply cluster-provision/centos9/Dockerfile:38]: If we talks about a new kernel default, it may be better to check for the kernel verson though ...

Comment (iholder101) [reply cluster-provision/centos9/Dockerfile:38]: I thought it should be under `cluster-provision/centos9` somewhere, not under the k8s scripts, but in second thought 1.27 is always being provisioned with centos9.

Thanks @rmohr! let me reshape the PR

**Comment by iholder101** [comment]
@rmohr Thanks again, PTAL

**Comment by rmohr** [comment]
/hold

feel free to unhold when you think this is ready.

**Review by xpivarc — COMMENTED** [review]
/lgtm
works for me

**Review by brianmcarey — COMMENTED** [review]
/lgtm

/unhold

**Pull request merged**